src/java/io/bazel/rulesscala/scalac/deps_tracking_reporter/after_2_13_12/DepsTrackingReporter.java

```diff
@@ -0,0 +1,320 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import scala.collection.immutable.List$;
+import scala.reflect.internal.util.CodeAction;
+import scala.reflect.internal.util.NoPosition$;
+import scala.reflect.internal.util.Position;
+import scala.tools.nsc.Settings;
+import scala.tools.nsc.reporters.ConsoleReporter;
+import scala.tools.nsc.reporters.Reporter;
+
+import javax.print.attribute.standard.Severity;
+
+public class DepsTrackingReporter extends ConsoleReporter {
+
+    private static final String HJAR_JAR_SUFFIX = "-hjar.jar";
+    private static final String IJAR_JAR_SUFFIX = "-ijar.jar";
+    private final Set<String> usedJars = new HashSet<>();
+
+    private final Map<String, String> jarToTarget = new HashMap<>();
+    private final Map<String, String> indirectJarToTarget = new HashMap<>();
+
+    private final Set<String> ignoredTargets;
+    private final Set<String> directTargets;
+
+    private final CompileOptions ops;
+    public final Reporter delegateReporter;
+    private Set<String> astUsedJars = new HashSet<>();
+
+    public DepsTrackingReporter(Settings settings, CompileOptions ops, Reporter delegate) {
+        super(settings);
+        this.ops = ops;
+        this.delegateReporter = delegate;
+
+        if (ops.directJars.length == ops.directTargets.length) {
+            for (int i = 0; i < ops.directJars.length; i++) {
+                jarToTarget.put(ops.directJars[i], ops.directTargets[i]);
+            }
+        } else {
+            throw new IllegalArgumentException(
+                    "mismatched size: directJars " + ops.directJars.length + " vs directTargets"
+                            + ops.directTargets.length);
+        }
+
+        if (ops.indirectJars.length == ops.indirectTargets.length) {
+            for (int i = 0; i < ops.indirectJars.length; i++) {
+                indirectJarToTarget.put(ops.indirectJars[i], ops.indirectTargets[i]);
+            }
+        } else {
+            throw new IllegalArgumentException(
+                    "mismatched size: indirectJars " + ops.directJars.length + " vs indirectTargets "
+                            + ops.directTargets.length);
+        }
+
+        ignoredTargets = Arrays.stream(ops.unusedDepsIgnoredTargets).collect(Collectors.toSet());
+        directTargets = Arrays.stream(ops.directTargets).collect(Collectors.toSet());
+    }
+
+    private boolean isDependencyTrackingOn() {
+        return "ast-plus".equals(ops.dependencyTrackingMethod)
+                && (!"off".equals(ops.strictDepsMode) || !"off".equals(ops.unusedDependencyCheckerMode));
+    }
+
+    @Override
+    public void doReport(Position pos, String msg, Severity severity, scala.collection.immutable.List<CodeAction> actions) {
```

Comment:
TLDR; After doing some troubleshooting, found out that the workaround was to override `doReport` instead of `info0` as mentioned [here](https://github.com/scala/scala/blob/v2.13.12/src/reflect/scala/reflect/internal/Reporting.scala#L103).

Long version : figured out that the `usedJars` were not populated properly in `DepsTrackingReporter` since the overridden `info0` method (responsible for building the list of user jars) didn't seem to be called. Turns out that the compiler patch that calls `inform` which in turns calls `echo` that now invokes `doReport`... so to get input about the used jars from the compiler source `doReport` needed to be overridden starting from Scala `2.13.12`...